### PR TITLE
Change project build system to poetry and minor bugfixes

### LINKED ISSUE #461: Cannot Install the `master` branch with Python 3.11.4
**Describe the bug**
I am unable to install the `master` branch with a Python 3.11.4 environment.

**To Reproduce**
1. Create and activate a Python 3.11 environment, I used `pyenv`, so I ran 
```python
pyenv virtualenv 3.11.4 mms
pyenv activate mms
```
2. Clone and install the repo with
```sh
git clone https://github.com/matchms/matchms.git
cd matchms
pip install .[dev]
```
**Expected behavior**

This does not occur when installing the latest version from PyPI with `pip install matchms`.

**Error Log**
<details><summary>Error Log</summary>

```python
Processing /Users/srivarra/Dev/matchms
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting deprecated (from matchms==0.21.1)
  Using cached Deprecated-1.2.14-py2.py3-none-any.whl (9.6 kB)
Collecting lxml (from matchms==0.21.1)
  Using cached lxml-4.9.3-cp311-cp311-macosx_11_0_universal2.whl (8.6 MB)
Collecting matplotlib (from matchms==0.21.1)
  Using cached matplotlib-3.7.2-cp311-cp311-macosx_11_0_arm64.whl (7.3 MB)
Collecting networkx (from matchms==0.21.1)
  Using cached networkx-3.1-py3-none-any.whl (2.1 MB)
Collecting numba>=0.47 (from matchms==0.21.1)
  Using cached numba-0.57.1-cp311-cp311-macosx_11_0_arm64.whl (2.5 MB)
Collecting numpy (from matchms==0.21.1)
  Using cached numpy-1.25.1-cp311-cp311-macosx_11_0_arm64.whl (14.0 MB)
Collecting pickydict>=0.4.0 (from matchms==0.21.1)
  Using cached pickydict-0.4.0-py3-none-any.whl (6.1 kB)
Collecting pillow!=9.4.0 (from matchms==0.21.1)
  Using cached Pillow-10.0.0-cp311-cp311-macosx_11_0_arm64.whl (3.1 MB)
Collecting pyteomics>=4.6 (from matchms==0.21.1)
  Using cached pyteomics-4.6-py2.py3-none-any.whl (235 kB)
Collecting requests (from matchms==0.21.1)
  Using cached requests-2.31.0-py3-none-any.whl (62 kB)
Collecting scipy (from matchms==0.21.1)
  Using cached scipy-1.11.1-cp311-cp311-macosx_12_0_arm64.whl (29.5 MB)
Collecting sparsestack>=0.4.1 (from matchms==0.21.1)
  Using cached sparsestack-0.4.1-py3-none-any.whl (10 kB)
Collecting tqdm (from matchms==0.21.1)
  Using cached tqdm-4.65.0-py3-none-any.whl (77 kB)
Collecting pandas (from matchms==0.21.1)
  Using cached pandas-2.0.3-cp311-cp311-macosx_11_0_arm64.whl (10.7 MB)
Collecting bump2version (from matchms==0.21.1)
  Using cached bump2version-1.0.1-py2.py3-none-any.whl (22 kB)
Collecting decorator (from matchms==0.21.1)
  Using cached decorator-5.1.1-py3-none-any.whl (9.1 kB)
Collecting isort>=5.1.0 (from matchms==0.21.1)
  Using cached isort-5.12.0-py3-none-any.whl (91 kB)
Collecting pylint<2.12 (from matchms==0.21.1)
  Using cached pylint-2.11.1-py3-none-any.whl (392 kB)
Collecting prospector[with_pyroma] (from matchms==0.21.1)
  Using cached prospector-1.10.2-py3-none-any.whl (77 kB)
Collecting pytest (from matchms==0.21.1)
  Using cached pytest-7.4.0-py3-none-any.whl (323 kB)
Collecting pytest-cov (from matchms==0.21.1)
  Using cached pytest_cov-4.1.0-py3-none-any.whl (21 kB)
Collecting sphinx>=4.0.0 (from matchms==0.21.1)
  Using cached sphinx-7.0.1-py3-none-any.whl (3.0 MB)
Collecting sphinx-rtd-theme (from matchms==0.21.1)
  Using cached sphinx_rtd_theme-1.2.2-py2.py3-none-any.whl (2.8 MB)
Collecting sphinxcontrib-apidoc (from matchms==0.21.1)
  Using cached sphinxcontrib_apidoc-0.3.0-py2.py3-none-any.whl (8.5 kB)
Collecting testfixtures (from matchms==0.21.1)
  Using cached testfixtures-7.1.0-py3-none-any.whl (102 kB)
Collecting yapf (from matchms==0.21.1)
  Using cached yapf-0.40.1-py3-none-any.whl (250 kB)
Collecting llvmlite<0.41,>=0.40.0dev0 (from numba>=0.47->matchms==0.21.1)
  Using cached llvmlite-0.40.1-cp311-cp311-macosx_11_0_arm64.whl (28.1 MB)
Collecting numpy (from matchms==0.21.1)
  Using cached numpy-1.24.4-cp311-cp311-macosx_11_0_arm64.whl (13.8 MB)
Collecting platformdirs>=2.2.0 (from pylint<2.12->matchms==0.21.1)
  Using cached platformdirs-3.9.0-py3-none-any.whl (16 kB)
Collecting astroid<2.9,>=2.8.0 (from pylint<2.12->matchms==0.21.1)
  Using cached astroid-2.8.6-py3-none-any.whl (248 kB)
Collecting mccabe<0.7,>=0.6 (from pylint<2.12->matchms==0.21.1)
  Using cached mccabe-0.6.1-py2.py3-none-any.whl (8.6 kB)
Collecting toml>=0.7.1 (from pylint<2.12->matchms==0.21.1)
  Using cached toml-0.10.2-py2.py3-none-any.whl (16 kB)
Collecting sphinxcontrib-applehelp (from sphinx>=4.0.0->matchms==0.21.1)
  Using cached sphinxcontrib_applehelp-1.0.4-py3-none-any.whl (120 kB)
Collecting sphinxcontrib-devhelp (from sphinx>=4.0.0->matchms==0.21.1)
  Using cached sphinxcontrib_devhelp-1.0.2-py2.py3-none-any.whl (84 kB)
Collecting sphinxcontrib-jsmath (from sphinx>=4.0.0->matchms==0.21.1)
  Using cached sphinxcontrib_jsmath-1.0.1-py2.py3-none-any.whl (5.1 kB)
Collecting sphinxcontrib-htmlhelp>=2.0.0 (from sphinx>=4.0.0->matchms==0.21.1)
  Using cached sphinxcontrib_htmlhelp-2.0.1-py3-none-any.whl (99 kB)
Collecting sphinxcontrib-serializinghtml>=1.1.5 (from sphinx>=4.0.0->matchms==0.21.1)
  Using cached sphinxcontrib_serializinghtml-1.1.5-py2.py3-none-any.whl (94 kB)
Collecting sphinxcontrib-qthelp (from sphinx>=4.0.0->matchms==0.21.1)
  Using cached sphinxcontrib_qthelp-1.0.3-py2.py3-none-any.whl (90 kB)
Collecting Jinja2>=3.0 (from sphinx>=4.0.0->matchms==0.21.1)
  Using cached Jinja2-3.1.2-py3-none-any.whl (133 kB)
Collecting Pygments>=2.13 (from sphinx>=4.0.0->matchms==0.21.1)
  Using cached Pygments-2.15.1-py3-none-any.whl (1.1 MB)
Collecting docutils<0.21,>=0.18.1 (from sphinx>=4.0.0->matchms==0.21.1)
  Using cached docutils-0.20.1-py3-none-any.whl (572 kB)
Collecting snowballstemmer>=2.0 (from sphinx>=4.0.0->matchms==0.21.1)
  Using cached snowballstemmer-2.2.0-py2.py3-none-any.whl (93 kB)
Collecting babel>=2.9 (from sphinx>=4.0.0->matchms==0.21.1)
  Using cached Babel-2.12.1-py3-none-any.whl (10.1 MB)
Collecting alabaster<0.8,>=0.7 (from sphinx>=4.0.0->matchms==0.21.1)
  Using cached alabaster-0.7.13-py3-none-any.whl (13 kB)
Collecting imagesize>=1.3 (from sphinx>=4.0.0->matchms==0.21.1)
  Using cached imagesize-1.4.1-py2.py3-none-any.whl (8.8 kB)
Collecting packaging>=21.0 (from sphinx>=4.0.0->matchms==0.21.1)
  Using cached packaging-23.1-py3-none-any.whl (48 kB)
Collecting charset-normalizer<4,>=2 (from requests->matchms==0.21.1)
  Using cached charset_normalizer-3.2.0-cp311-cp311-macosx_11_0_arm64.whl (122 kB)
Collecting idna<4,>=2.5 (from requests->matchms==0.21.1)
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Collecting urllib3<3,>=1.21.1 (from requests->matchms==0.21.1)
  Using cached urllib3-2.0.3-py3-none-any.whl (123 kB)
Collecting certifi>=2017.4.17 (from requests->matchms==0.21.1)
  Using cached certifi-2023.5.7-py3-none-any.whl (156 kB)
Collecting wrapt<2,>=1.10 (from deprecated->matchms==0.21.1)
  Using cached wrapt-1.15.0-cp311-cp311-macosx_11_0_arm64.whl (36 kB)
Collecting contourpy>=1.0.1 (from matplotlib->matchms==0.21.1)
  Using cached contourpy-1.1.0-cp311-cp311-macosx_11_0_arm64.whl (229 kB)
Collecting cycler>=0.10 (from matplotlib->matchms==0.21.1)
  Using cached cycler-0.11.0-py3-none-any.whl (6.4 kB)
Collecting fonttools>=4.22.0 (from matplotlib->matchms==0.21.1)
  Using cached fonttools-4.41.0-cp311-cp311-macosx_10_9_universal2.whl (2.5 MB)
Collecting kiwisolver>=1.0.1 (from matplotlib->matchms==0.21.1)
  Using cached kiwisolver-1.4.4-cp311-cp311-macosx_11_0_arm64.whl (63 kB)
Collecting pyparsing<3.1,>=2.3.1 (from matplotlib->matchms==0.21.1)
  Using cached pyparsing-3.0.9-py3-none-any.whl (98 kB)
Collecting python-dateutil>=2.7 (from matplotlib->matchms==0.21.1)
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting pytz>=2020.1 (from pandas->matchms==0.21.1)
  Using cached pytz-2023.3-py2.py3-none-any.whl (502 kB)
Collecting tzdata>=2022.1 (from pandas->matchms==0.21.1)
  Using cached tzdata-2023.3-py2.py3-none-any.whl (341 kB)
Collecting GitPython<4.0.0,>=3.1.27 (from prospector[with_pyroma]->matchms==0.21.1)
  Using cached GitPython-3.1.32-py3-none-any.whl (188 kB)
Collecting PyYAML (from prospector[with_pyroma]->matchms==0.21.1)
  Using cached PyYAML-6.0-cp311-cp311-macosx_11_0_arm64.whl (167 kB)
Collecting dodgy<0.3.0,>=0.2.1 (from prospector[with_pyroma]->matchms==0.21.1)
  Using cached dodgy-0.2.1-py3-none-any.whl (5.4 kB)
Collecting flake8<6.0.0 (from prospector[with_pyroma]->matchms==0.21.1)
  Using cached flake8-5.0.4-py2.py3-none-any.whl (61 kB)
INFO: pip is looking at multiple versions of prospector[with-pyroma] to determine which version is compatible with other requirements. This could take a while.
Collecting prospector[with_pyroma] (from matchms==0.21.1)
  Using cached prospector-1.10.0-py3-none-any.whl (87 kB)
  Using cached prospector-1.9.0-py3-none-any.whl (85 kB)
  Using cached prospector-1.8.4-py3-none-any.whl (85 kB)
  Using cached prospector-1.8.3-py3-none-any.whl (85 kB)
  Using cached prospector-1.8.2-py3-none-any.whl (85 kB)
WARNING: prospector 1.8.2 does not provide the extra 'with_pyroma'
  Using cached prospector-1.8.1-py3-none-any.whl (85 kB)
WARNING: prospector 1.8.1 does not provide the extra 'with_pyroma'
  Using cached prospector-1.8.0-py3-none-any.whl (85 kB)
INFO: pip is looking at multiple versions of prospector[with-pyroma] to determine which version is compatible with other requirements. This could take a while.
  Using cached prospector-1.7.7-py3-none-any.whl (85 kB)
Collecting pep8-naming<=0.10.0,>=0.3.3 (from prospector[with_pyroma]->matchms==0.21.1)
  Using cached pep8_naming-0.10.0-py2.py3-none-any.whl (8.1 kB)
Collecting pycodestyle<2.9.0,>=2.6.0 (from prospector[with_pyroma]->matchms==0.21.1)
  Using cached pycodestyle-2.8.0-py2.py3-none-any.whl (42 kB)
Collecting pydocstyle>=2.0.0 (from prospector[with_pyroma]->matchms==0.21.1)
  Using cached pydocstyle-6.3.0-py3-none-any.whl (38 kB)
Collecting pyflakes<3,>=2.2.0 (from prospector[with_pyroma]->matchms==0.21.1)
  Using cached pyflakes-2.5.0-py2.py3-none-any.whl (66 kB)
Collecting pylint-celery==0.3 (from prospector[with_pyroma]->matchms==0.21.1)
  Using cached pylint_celery-0.3-py3-none-any.whl
Collecting pylint-django<2.6,>=2.5 (from prospector[with_pyroma]->matchms==0.21.1)
  Using cached pylint_django-2.5.3-py3-none-any.whl (81 kB)
Collecting pylint-flask==0.6 (from prospector[with_pyroma]->matchms==0.21.1)
  Using cached pylint_flask-0.6-py3-none-any.whl
Collecting pylint-plugin-utils<0.8,>=0.7 (from prospector[with_pyroma]->matchms==0.21.1)
  Using cached pylint_plugin_utils-0.7-py3-none-any.whl (10 kB)
Collecting pyroma>=2.4 (from prospector[with_pyroma]->matchms==0.21.1)
  Using cached pyroma-4.2-py3-none-any.whl (22 kB)
Collecting requirements-detector<0.8,>=0.7 (from prospector[with_pyroma]->matchms==0.21.1)
  Using cached requirements-detector-0.7.tar.gz (8.6 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting setoptconf-tmp<0.4.0,>=0.3.1 (from prospector[with_pyroma]->matchms==0.21.1)
  Using cached setoptconf_tmp-0.3.1-py3-none-any.whl (32 kB)
Collecting iniconfig (from pytest->matchms==0.21.1)
  Using cached iniconfig-2.0.0-py3-none-any.whl (5.9 kB)
Collecting pluggy<2.0,>=0.12 (from pytest->matchms==0.21.1)
  Using cached pluggy-1.2.0-py3-none-any.whl (17 kB)
Collecting coverage[toml]>=5.2.1 (from pytest-cov->matchms==0.21.1)
  Using cached coverage-7.2.7-cp311-cp311-macosx_11_0_arm64.whl (201 kB)
Collecting sphinx>=4.0.0 (from matchms==0.21.1)
  Using cached sphinx-6.2.1-py3-none-any.whl (3.0 MB)
Collecting docutils<0.21,>=0.18.1 (from sphinx>=4.0.0->matchms==0.21.1)
  Using cached docutils-0.18.1-py2.py3-none-any.whl (570 kB)
Collecting sphinxcontrib-jquery<5,>=4 (from sphinx-rtd-theme->matchms==0.21.1)
  Using cached sphinxcontrib_jquery-4.1-py2.py3-none-any.whl (121 kB)
Collecting pbr (from sphinxcontrib-apidoc->matchms==0.21.1)
  Using cached pbr-5.11.1-py2.py3-none-any.whl (112 kB)
Collecting importlib-metadata>=6.6.0 (from yapf->matchms==0.21.1)
  Using cached importlib_metadata-6.8.0-py3-none-any.whl (22 kB)
Collecting tomli>=2.0.1 (from yapf->matchms==0.21.1)
  Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting lazy-object-proxy>=1.4.0 (from astroid<2.9,>=2.8.0->pylint<2.12->matchms==0.21.1)
  Using cached lazy_object_proxy-1.9.0-cp311-cp311-macosx_13_0_arm64.whl
Collecting wrapt<2,>=1.10 (from deprecated->matchms==0.21.1)
  Using cached wrapt-1.13.3.tar.gz (48 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'error'
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [86 lines of output]
      Traceback (most recent call last):
        File "/private/var/folders/fy/q2szypn9325d_0g0nq049k300000gq/T/pip-build-env-ic6o58ce/overlay/lib/python3.11/site-packages/setuptools/config/expand.py", line 81, in __getattr__
          return next(
                 ^^^^^
        File "/private/var/folders/fy/q2szypn9325d_0g0nq049k300000gq/T/pip-build-env-ic6o58ce/overlay/lib/python3.11/site-packages/setuptools/config/expand.py", line 82, in <genexpr>
          ast.literal_eval(value)
        File "/Users/srivarra/.pyenv/versions/3.11.4/lib/python3.11/ast.py", line 110, in literal_eval
          return _convert(node_or_string)
                 ^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/srivarra/.pyenv/versions/3.11.4/lib/python3.11/ast.py", line 109, in _convert
          return _convert_signed_num(node)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/srivarra/.pyenv/versions/3.11.4/lib/python3.11/ast.py", line 83, in _convert_signed_num
          return _convert_num(node)
                 ^^^^^^^^^^^^^^^^^^
        File "/Users/srivarra/.pyenv/versions/3.11.4/lib/python3.11/ast.py", line 74, in _convert_num
          _raise_malformed_node(node)
        File "/Users/srivarra/.pyenv/versions/3.11.4/lib/python3.11/ast.py", line 71, in _raise_malformed_node
          raise ValueError(msg + f': {node!r}')
      ValueError: malformed node or string on line 2: <ast.Call object at 0x1055ac040>
      
      The above exception was the direct cause of the following exception:
      
      Traceback (most recent call last):
        File "/private/var/folders/fy/q2szypn9325d_0g0nq049k300000gq/T/pip-build-env-ic6o58ce/overlay/lib/python3.11/site-packages/setuptools/config/expand.py", line 191, in read_attr
          return getattr(StaticModule(module_name, spec), attr_name)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/fy/q2szypn9325d_0g0nq049k300000gq/T/pip-build-env-ic6o58ce/overlay/lib/python3.11/site-packages/setuptools/config/expand.py", line 87, in __getattr__
          raise AttributeError(f"{self.name} has no attribute {attr}") from e
      AttributeError: wrapt has no attribute __version__
      
      During handling of the above exception, another exception occurred:
      
      Traceback (most recent call last):
        File "/Users/srivarra/.pyenv/versions/3.11.4/envs/mms/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/Users/srivarra/.pyenv/versions/3.11.4/envs/mms/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/srivarra/.pyenv/versions/3.11.4/envs/mms/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 149, in prepare_metadata_for_build_wheel
          return hook(metadata_directory, config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/fy/q2szypn9325d_0g0nq049k300000gq/T/pip-build-env-ic6o58ce/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 380, in prepare_metadata_for_build_wheel
          self.run_setup()
        File "/private/var/folders/fy/q2szypn9325d_0g0nq049k300000gq/T/pip-build-env-ic6o58ce/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 488, in run_setup
          self).run_setup(setup_script=setup_script)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/fy/q2szypn9325d_0g0nq049k300000gq/T/pip-build-env-ic6o58ce/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 338, in run_setup
          exec(code, locals())
        File "<string>", line 41, in <module>
        File "/private/var/folders/fy/q2szypn9325d_0g0nq049k300000gq/T/pip-build-env-ic6o58ce/overlay/lib/python3.11/site-packages/setuptools/__init__.py", line 107, in setup
          return distutils.core.setup(**attrs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/fy/q2szypn9325d_0g0nq049k300000gq/T/pip-build-env-ic6o58ce/overlay/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 159, in setup
          dist.parse_config_files()
        File "/private/var/folders/fy/q2szypn9325d_0g0nq049k300000gq/T/pip-build-env-ic6o58ce/overlay/lib/python3.11/site-packages/setuptools/dist.py", line 894, in parse_config_files
          setupcfg.parse_configuration(
        File "/private/var/folders/fy/q2szypn9325d_0g0nq049k300000gq/T/pip-build-env-ic6o58ce/overlay/lib/python3.11/site-packages/setuptools/config/setupcfg.py", line 189, in parse_configuration
          meta.parse()
        File "/private/var/folders/fy/q2szypn9325d_0g0nq049k300000gq/T/pip-build-env-ic6o58ce/overlay/lib/python3.11/site-packages/setuptools/config/setupcfg.py", line 500, in parse
          section_parser_method(section_options)
        File "/private/var/folders/fy/q2szypn9325d_0g0nq049k300000gq/T/pip-build-env-ic6o58ce/overlay/lib/python3.11/site-packages/setuptools/config/setupcfg.py", line 475, in parse_section
          self[name] = value
          ~~~~^^^^^^
        File "/private/var/folders/fy/q2szypn9325d_0g0nq049k300000gq/T/pip-build-env-ic6o58ce/overlay/lib/python3.11/site-packages/setuptools/config/setupcfg.py", line 293, in __setitem__
          parsed = self.parsers.get(option_name, lambda x: x)(value)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/fy/q2szypn9325d_0g0nq049k300000gq/T/pip-build-env-ic6o58ce/overlay/lib/python3.11/site-packages/setuptools/config/setupcfg.py", line 606, in _parse_version
          return expand.version(self._parse_attr(value, self.package_dir, self.root_dir))
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/fy/q2szypn9325d_0g0nq049k300000gq/T/pip-build-env-ic6o58ce/overlay/lib/python3.11/site-packages/setuptools/config/setupcfg.py", line 417, in _parse_attr
          return expand.read_attr(attr_desc, package_dir, root_dir)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/fy/q2szypn9325d_0g0nq049k300000gq/T/pip-build-env-ic6o58ce/overlay/lib/python3.11/site-packages/setuptools/config/expand.py", line 194, in read_attr
          module = _load_spec(spec, module_name)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/fy/q2szypn9325d_0g0nq049k300000gq/T/pip-build-env-ic6o58ce/overlay/lib/python3.11/site-packages/setuptools/config/expand.py", line 214, in _load_spec
          spec.loader.exec_module(module)  # type: ignore
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "<frozen importlib._bootstrap_external>", line 940, in exec_module
        File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
        File "/private/var/folders/fy/q2szypn9325d_0g0nq049k300000gq/T/pip-install-8dead4lw/wrapt_6c365039e7fb4efa9672b80c7aec43ea/src/wrapt/__init__.py", line 10, in <module>
          from .decorators import (adapter_factory, AdapterFactory, decorator,
        File "/private/var/folders/fy/q2szypn9325d_0g0nq049k300000gq/T/pip-install-8dead4lw/wrapt_6c365039e7fb4efa9672b80c7aec43ea/src/wrapt/decorators.py", line 34, in <module>
          from inspect import ismethod, isclass, formatargspec
      ImportError: cannot import name 'formatargspec' from 'inspect' (/Users/srivarra/.pyenv/versions/3.11.4/lib/python3.11/inspect.py)
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

[notice] A new release of pip is available: 23.1.2 -> 23.2
[notice] To update, run: python3.11 -m pip install --upgrade pip

```
</details>

**Desktop (please complete the following information):**
 - OS: macOS Ventura 13.4.1 (c)

**Additional context**

This occurs because the the package `pylint<0.12` in the dev extras cannot use later versions of `wrapt`, which `deprecated` requires. 
Found here:
https://github.com/matchms/matchms/blob/b1fcb7f50954ce84e4292e393305ee508e515259/setup.py#L66C37-L66C37

I'd love to contribute a fix if possible!

### LINKED ISSUE #462: Error loading MoNA msp file, multiple warnings regarding inability to convert retention time to float
Downloaded MoNA msp files (specifically the LC-MS/MS orbitrap dataset) yields the following warnings. All mostly related to the inability to convert retention time to float:

`
2023-07-24 16:37:25,619:WARNING:matchms:add_precursor_mz:No precursor_mz found in metadata.
2023-07-24 16:37:25,619:WARNING:matchms:add_retention:4.713093 min can't be converted to float.
2023-07-24 16:37:25,620:WARNING:matchms:add_retention:4.813797 min can't be converted to float.
2023-07-24 16:37:25,620:WARNING:matchms:add_retention:4.810467 min can't be converted to float.
2023-07-24 16:37:25,624:WARNING:matchms:add_precursor_mz:No precursor_mz found in metadata.
2023-07-24 16:37:25,624:WARNING:matchms:add_retention:4.799763 min can't be converted to float.
2023-07-24 16:37:25,625:WARNING:matchms:add_retention:4.623643 min can't be converted to float.
2023-07-24 16:37:25,625:WARNING:matchms:add_retention:4.620587 min can't be converted to float.
2023-07-24 16:37:25,626:WARNING:matchms:add_precursor_mz:No precursor_mz found in metadata.
2023-07-24 16:37:25,626:WARNING:matchms:add_retention:4.612950 min can't be converted to float.
2023-07-24 16:37:25,627:WARNING:matchms:add_retention:5.072005 min can't be converted to float.
2023-07-24 16:37:25,627:WARNING:matchms:add_retention:5.064583 min can't be converted to float.
2023-07-24 16:37:25,628:WARNING:matchms:add_precursor_mz:No precursor_mz found in metadata.
2023-07-24 16:37:25,628:WARNING:matchms:add_retention:5.057092 min can't be converted to float.
2023-07-24 16:37:25,629:WARNING:matchms:add_retention:4.570043 min can't be converted to float.
2023-07-24 16:37:25,629:WARNING:matchms:add_retention:4.567235 min can't be converted to float.
2023-07-24 16:37:25,630:WARNING:matchms:add_precursor_mz:No precursor_mz found in metadata.
2023-07-24 16:37:25,630:WARNING:matchms:add_retention:4.559692 min can't be converted to float.
`

Clearly the retention time is present in the file but is not being parsed correctly. I believe that the no precursor mz warning is correct. The few examples I checked are because they are in-source fragments.  

I am on OSX using an M2 Mac Book Pro. Most recent version of matchms.

### LINKED ISSUE #463: `pyyaml` could be a core dependency
I again made a fresh environment via:

```shell
mkdir matchms-test
poetry init # set python 3.9 as default
poetry add matchms
```

```python
>>> import matchms

Configuration file exists at /Users/shaurya/Library/Preferences/pypoetry, reusing this directory.

Consider moving TOML configuration files to /Users/shaurya/Library/Application Support/pypoetry, as support for the legacy directory will be removed in an upcoming release.
Python 3.9.16 (main, Apr 24 2023, 11:18:10)
[Clang 14.0.3 (clang-1403.0.22.14.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import matchms
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/shaurya/code/matchms-test/.venv/lib/python3.9/site-packages/matchms/__init__.py", line 8, in <module>
    from .Pipeline import Pipeline
  File "/Users/shaurya/code/matchms-test/.venv/lib/python3.9/site-packages/matchms/Pipeline.py", line 5, in <module>
    import yaml
ModuleNotFoundError: No module named 'yaml'
>>>
```

doing a `poetry add pyyaml` fixes it. Perhaps we could add `pyyaml` as a core dependency.

_Originally posted by @chanana in https://github.com/matchms/matchms/issues/456#issuecomment-1649929003_

### LINKED ISSUE #464: Change build system to use poetry
Using poetry as a build system and a `pyproject.toml` file might reduce some of the overhead related to dependencies and might make testing and overall setup easier.

### PR DESCRIPTION
**Pull request by hechth**
This PR overhauls the matchms project structure.

I've removed `setup.py` and `setup.cfg` and replaced them with the `pyproject.toml` file, as is recommended by the python community. The `pyproject.toml` file is configured with poetry to install dependencies, run tests and build the software package.

This includes several version updates and changes and also retires python 3.7 as it has reached it's end of life.
Python 3.8 is so far supported, I also added Python 3.11 to the roster. I also added missing dependencies (pyyaml).

Linting errors reported by pylint have been fixed - don't know how they went unnoticed so far.

This pull request also adds RDKit as a normal dependency - it seems to be all working fine so I don't see a reason to not use it, since we also now added pandas to the list of dependencies.

@florian-huber if this is okay for you I would also then clean up the rdkit chem related stuff in this or a follow-up PR.

Closes #464 
Closes #462 
Closes #463 
Closes #461

### CODE REVIEW

**Comment by niekdejonge** [comment]
Looks good. I have no experience with Poetry, but might be a good step. Would the benefit be speed up of running the tests?

**Comment by niekdejonge** [comment]
Regarding the RDKit dependency. The rationale was to have as few large dependencies as possible, so others can use Matchms without depending on large packages, making installing faster, but mostly to reduce the risk of having dependency issues. I think this really is something that is useful since we expect others to build matchsms into other packages. However, RDKit is now used quite a bit throughout matchms. The checks we now do to check if rdkit is installed and the fact that we have to use a different test pipeline for rdkit, makes development slower and the code base less clean, which to me might outweigh the benefit of having less large dependencies. I don't have any clear preference. What do you think @florian-huber ?

**Review by niekdejonge — APPROVED** [review]
See my other comments for some discussion, but looks good! I had a quick look through the linting fixes and they make sense to me, thanks!

File: .github/workflows/CI_build.yml [hunk 1]
```diff
@@ -77,26 +77,26 @@ jobs:
...
       - name: Test package
         run: |
+          poetry install --only dev
           python -m twine check dist/*
       - name: Show pip list
         run: |
           pip list
       - name: Install development dependencies
         run: |
-          pip install -e .[dev]
```
Comment (florian-huber) [inline .github/workflows/CI_build.yml:93]: In the readme part for developers it still reads: `pip install --editable .[dev]`
I guess that would no longer work then, right?

**Review by florian-huber — APPROVED** [review]
@hechth nice work!
Looks good to me (except minor edit required in the readme).

**Comment by niekdejonge** [comment]
If we add rdkit to the standard requirements, the pytest.importorskip("rdkit") can be removed from the tests and the checks if rdkit is installed can also be removed at some point. Might also be fine to do this in a different pull request.

File: .github/workflows/CI_build.yml (hunk 1 above)
Comment (hechth) [reply .github/workflows/CI_build.yml:93]: I think it does - let me check!

Comment (hechth) [reply .github/workflows/CI_build.yml:93]: seems to be still working fine!

Comment (florian-huber) [reply .github/workflows/CI_build.yml:93]: Ok, even better! Thanks!

**Pull request merged**